pxtblocks/fields/field_melodySandbox.ts

```diff
@@ -0,0 +1,415 @@
+/// <reference path="../../built/pxtlib.d.ts" />
+
+namespace pxtblockly {
+    export class FieldCustomMelody<U extends Blockly.FieldCustomOptions> extends Blockly.Field implements Blockly.FieldCustom {
+        public isFieldCustom_ = true;
+        protected params: U;
+        private title: string = "Name this tune";
+        private melody: pxtmelody.MelodyArray;
+        private soundingKeys: number = 0;
+        private numRow: number = 8;
+        private numCol: number = 8;
+        private tempo: number = 120;
+        private stringRep: string;
+        private oneNotePerCol: boolean = true;
+        private isPlaying: boolean = false;
+        private isLooping: boolean = false;
+        private timeouts: any = []; // keep track of timeouts
```

Comment:
Don't use `any` if possible

Reply:
I'm using this to keep track of the setTimeout calls. Is there a way to define it as an array of function calls?

Reply:
setTimeout and setInterval both return an ID number: https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#Return_value